### PR TITLE
Support GORA APIs

### PR DESCRIPTION
**Pull request by kamatama41**
issue: https://github.com/rakuten-ws/rws-ruby-sdk/issues/28

@satoryu 
Please review, thank you.

### CODE REVIEW

**Comment by satoryu** [comment]
For all supported versions of ruby, the following example could not be passed. 
https://travis-ci.org/rakuten-ws/rws-ruby-sdk/jobs/137737357#L486-L502

```
 1) RakutenWebService::Gora::Course.search just call the search method after that, call each endpoint should be called

     Failure/Error: expect(@second_request).to have_been_made.once

       The request GET https://app.rakuten.co.jp/services/api/Gora/GoraGolfCourseSearch/20131113?affiliateId=dummy_affiliate_id&applicationId=dummy_application_id&keyword=%E8%BB%BD%E4%BA%95%E6%B2%A2&page=2 was expected to execute 1 time but it executed 0 times

       The following requests were made:

       GET https://app.rakuten.co.jp/services/api/Gora/GoraGolfCourseSearch/20131113?affiliateId=dummy_affiliate_id&applicationId=dummy_application_id&keyword=%E8%BB%BD%E4%BA%95%E6%B2%A2 with headers {'Accept'=>'*/*', 'Accept-Encoding'=>'gzip;q=1.0,deflate;q=0.6,identity;q=0.3', 'User-Agent'=>'RakutenWebService SDK for Ruby v0.6.3(ruby-2.1.5 [x86_64-linux])'} was made 1 time

       ============================================================

     # ./spec/rakuten_web_service/gora/course_spec.rb:71:in `block (5 levels) in <top (required)>'

Finished in 0.80172 seconds

102 examples, 1 failure

Failed examples:

rspec ./spec/rakuten_web_service/gora/course_spec.rb:69 [36m# RakutenWebService::Gora::Course.search just call the search method after that, call each endpoint should be called
```

**Comment by kamatama41** [comment]
Umm, some tests are failed..
will have a look.

File: spec/rakuten_web_service/gora/course_detail_spec.rb [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  let(:expected_query) do
+    {
+        :affiliateId => affiliate_id,
+        :applicationId => application_id,
+        :golfCourseId => 120092
+    }
+  end
+
+  before do
+    RakutenWebService.configuration do |c|
```
Comment (satoryu) [inline spec/rakuten_web_service/gora/course_detail_spec.rb]: Calling `RWS.configuration` with a block will be deprecated from the next release. please use `RWS.configure`.

File: spec/rakuten_web_service/gora/course_detail_spec.rb [hunk 2]
```diff
@@ -0,0 +1,61 @@
+# encoding: utf-8
+
+require 'spec_helper'
+require 'rakuten_web_service'
```
Comment (satoryu) [inline spec/rakuten_web_service/gora/course_detail_spec.rb]: please remove this `require` from all spec files. `require`ing `rakuten_web_service` in spec files causes SimpleCov fail to measure coverages. I got known it in this week...

File: spec/rakuten_web_service/gora/course_spec.rb [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+  describe '.search' do
+    before do
+      response = JSON.parse(fixture('gora/course_search_with_Karuizawa.json'))
+      @expected_request = stub_request(:get, endpoint).
+          with(:query => expected_query).to_return(:body => response.to_json)
+
+      response['page'] = 2
+      response['first'] = 31
+      response['last'] = 60
+      @second_request = stub_request(:get, endpoint).
```
Comment (satoryu) [inline spec/rakuten_web_service/gora/course_spec.rb]: I got the root cause of [the error](https://travis-ci.org/rakuten-ws/rws-ruby-sdk/jobs/137737357#L486-L502). This is caused by the change I merged recently. 

Now  `SearchResult#each` does not pagerize automatically but provides only resources given from a response. 
So you can remove this stub_request.

Comment (kamatama41) [inline spec/rakuten_web_service/gora/course_spec.rb]: Ah, got it.
So I fixed the test to expect not to be called `@second_request` and added a test for `#all`

File: spec/rakuten_web_service/gora/course_detail_spec.rb (hunk 1 above)
Comment (kamatama41) [inline spec/rakuten_web_service/gora/course_detail_spec.rb]: Fixed at a04599b

File: spec/rakuten_web_service/gora/course_detail_spec.rb (hunk 2 above)
Comment (kamatama41) [inline spec/rakuten_web_service/gora/course_detail_spec.rb]: removed at 1064e34

**Pull request merged**